velox/expression/CastExpr.cpp

```diff
@@ -476,11 +440,6 @@ VectorPtr CastExpr::applyRow(
     const auto& toChildType = toRowType.childAt(toChildrenIndex);
 
     if (matchNotFound) {
-      if (nullOnFailure_) {
-        VELOX_USER_FAIL(
-            "Invalid complex cast the match is not found for the field {}",
-            toFieldName)
-      }
```

Comment:
I'm wondering why the original code throws when nullOnFailure_ is true and set nulls when this flag is false. This feels counterintuitive. I'd expect the opposite.